docs/src/adr/002adr-types.md

```diff
@@ -149,45 +229,55 @@ The parser only supports one-line comments that starts with `//`. Since type
 without having algebraic datatypes in TLA+. It also reminds the user that one
 better tests the field `type` carefully.
 
-Type System 1 is also very much in line with the [type system by Stephan Merz and Hernan Vanzetto](https://dblp.org/search?q=Automatic+Verification+of+%7BTLA%7D+%2B+Proof+Obligations+with+%7BSMT%7D+Solvers)
-, which is used internally by
-[TLAPS](https://tla.msr-inria.inria.fr/tlaps/content/Home.html) when translating proof obligations in SMT. We introduce
-types for user-defined operators, on top of their types for TLA+ expressions that do not contain user-defined operators.
-
-We expect that this type system will evolve in the future. That is why we call it __Type System 1__. Feel free to
-suggest __Type System 2__ :-)
-
-**Note:** For the above example of a set of records, we are considering to introduce union types. So the type of the set
+**Sets of records in Type System 1.2.** Consider the following set:
 
 ```tla
-{[type |-> "1a", bal |-> 1],
- [type |-> "2a", bal |-> 2, val |-> 3]}
+{[tag |-> "1a", bal |-> 1],
+ [tag |-> "2a", bal |-> 2, val |-> 3]}
 ```
 
-would be something like:
+In Type System 1.2 (Section 1.3), this set has the type of a set over a variant
+type:
 
 ```tla
-Set([type |-> "1a"], bal |-> 1]
-  + [type |-> "2a", bal |-> 2, val |-> 3])
+Set({ tag: "1a", bal: Int } | { tag: "2a", bal: Int, val: Int })
 ```
 
-The value of the field `type` would serve as a type tag. However, we would have to fix a set of patterns that turn a
-union type into a precise record type. One such pattern is a set comprehension, e.g., `{ r \in S: r.type = "1a" }`. If
-you have suggestions on this line of thought, please let us know.
+The value of the field `tag` serves as a type tag. However, we have to fix a
+set of patterns that turn a variant type into a precise record type. In untyped
+TLA+, such pattern is a set comprehension, e.g., `{ r \in S: r.tag = "1a" }`.
+In the typed version, we define a minimal set of operators over variants in the
+module
+[Variants.tla](https://github.com/informalsystems/apalache/blob/unstable/src/tla/Variants.tla).
+For instance, instead of writing the set comprehension, we have to use a filter
+over a set of variants: `FilterByTag(S, "1a")`.
+
+**Other type systems.**
+Type System 1 is also very much in line with the [type system by Stephan Merz and Hernan Vanzetto](https://dblp.org/search?q=Automatic+Verification+of+%7BTLA%7D+%2B+Proof+Obligations+with+%7BSMT%7D+Solvers)
+, which is used internally by
+[TLAPS](https://tla.msr-inria.inria.fr/tlaps/content/Home.html) when translating proof obligations in SMT. We introduce
+types for user-defined operators, on top of their types for TLA+ expressions that do not contain user-defined operators.
+
+We expect that this type system will evolve in the future. That is why we call
+it __Type System 1__. Section 1.3 presents its extension to __Type System
+1.2__. Feel free to suggest __Type System 2.0__ :-)
```

Comment:
Was there a Type System 1.1?

Reply:
Nope, we are going directly from 1 to 1.2 :))

Reply:
Why? Isn't this confusing?

Reply:
May be. I don't find it really confusing though. The most important property of versions is monotonicity.